### PR TITLE
separate callback_group from subscription options

### PR DESCRIPTION
**Pull request by fujitatomoya**
aligned with https://github.com/ros2/rclcpp/pull/1833

Signed-off-by: Tomoya Fujita <Tomoya.Fujita@sony.com>

### CODE REVIEW

**Comment by fujitatomoya** [comment]
@iuhilnehc-ynos requesting review.

File: rclcpp/executors/multithreaded_executor/multithreaded_executor.cpp [hunk 1]
```diff
@@ -101,8 +99,9 @@ class DualThreadedNode : public rclcpp::Node
                                             // So we just put a generic placeholder
                                             // into the binder
                                             // (since we know we need ONE parameter)
-      sub1_opt);                  // This is where we set the callback group.
-                                  // This subscription will run with callback group subscriber1
+      sub1_opt,
+      callback_group_subscriber1_);         // This is where we set the callback group.
+                                            // This subscription will run with callback group subscriber1
```
Comment (iuhilnehc-ynos) [inline rclcpp/executors/multithreaded_executor/multithreaded_executor.cpp:104]: Maybe you have noticed `Lines should be <= 100 characters long` in the CI warning.

File: rclcpp/topics/minimal_subscriber/static_wait_set_subscriber.cpp [hunk 2]
```diff
@@ -35,15 +35,15 @@ class StaticWaitSetSubscriber : public rclcpp::Node
         rclcpp::CallbackGroup::SharedPtr cb_group_waitset = this->create_callback_group(
```
Comment (iuhilnehc-ynos) [inline rclcpp/topics/minimal_subscriber/static_wait_set_subscriber.cpp:35]: I think this `cb_group_waitset` should be kept its lifecycle. What about putting it in the `StaticWaitSetSubscriber::cb_group_waitset_`? Same for the `rclcpp/topics/minimal_subscriber/time_triggered_wait_set_subscriber.cpp`, `rclcpp/topics/minimal_subscriber/wait_set_subscriber.cpp` and `rclcpp/wait_set/src/listener.cpp`.

**Review by iuhilnehc-ynos — CHANGES_REQUESTED** [review]
Some minor comments.

File: rclcpp/executors/multithreaded_executor/multithreaded_executor.cpp (hunk 1 above)
Comment (fujitatomoya) [reply rclcpp/executors/multithreaded_executor/multithreaded_executor.cpp]: nice catch 👍 done

File: rclcpp/topics/minimal_subscriber/static_wait_set_subscriber.cpp (hunk 2 above)
Comment (fujitatomoya) [reply rclcpp/topics/minimal_subscriber/static_wait_set_subscriber.cpp:35]: right, example node instance should keep the ownership of CallbackGroup.

Comment (fujitatomoya) [reply rclcpp/topics/minimal_subscriber/static_wait_set_subscriber.cpp]: fixed in https://github.com/ros2/examples/commit/1fc83d440124d76cf23b97f4f4ed5742d61faf95

**Comment by fujitatomoya** [comment]
@clalancette @adityapande-1995 could you do review on this?

File: rclcpp/topics/minimal_subscriber/static_wait_set_subscriber.cpp (hunk 2 above)
Comment (iuhilnehc-ynos) [reply rclcpp/topics/minimal_subscriber/static_wait_set_subscriber.cpp]: Thanks for the update. Could you tell me why not update the `rclcpp/wait_set/src/listener.cpp`?

Comment (fujitatomoya) [reply rclcpp/topics/minimal_subscriber/static_wait_set_subscriber.cpp]: Ah 😅 that was just an oversight. thanks. i will fix it accordingly.

**Comment by fujitatomoya** [comment]
@ralph-lange @sloretz requesting another review.

**Review by ralph-lange — APPROVED** [review]
LGTM. Can be merged as soon as https://github.com/ros2/rclcpp/pull/1833 has been merged.

**Review by wjwwood — APPROVED** [review]

**Comment by fujitatomoya** [comment]
closing because of https://github.com/ros2/rclcpp/pull/1833#issuecomment-1131095686

**Pull request closed without merging**